### DISCUSSION TITLE
Using `dcterms:conformsTo` as `onProperty` in restriction.

### DISCUSSION THREAD

**Discussion by AndreasMeier12** [apache/jena#2815] [Q&A] [answered]
I can't use `DCTerms.conformsTo()` in a restriction.
I am working with a file that specifies this property in a restriction. 
Restriction in TTL format: 
``` 
[ a owl:Restriction ;
    owl:minCardinality "1"^^xsd:nonNegativeInteger ;
    owl:onProperty dcterms:conformsTo ]

``` 
This is parsed but I cannot do anything useful as `restriction.getOnProperty()` throws a ConversionException. From debugging into JENA, I have found that this usage is not supported. 
`org.apache.jena.ontology.ConversionException: Cannot convert node http://purl.org/dc/terms/conformsTo to OntProperty` 

Is this an issue in the way that I use JENA? If not,  aside from working around this, is there anyhting I can do to ? 

Is there anything else that could help answer this question? 

I am using JENA version 5.0.0

**Comment by afs** [comment]
Hi @AndreasMeier12

Jena now has a more comprehensive ontology API , introduced in Jena 5.1.0.

https://jena.apache.org/documentation/ontology/
There is a maven artifact `org.apache.jena:jena-ontapi`

It is would be helpful if you can provide a complete, minimal example (code) to illustrate the issue.

**Comment by AndreasMeier12** [comment]
Thank you for your answer @afs.
I have created an example: `https://github.com/AndreasMeier12/jena-dc-ontology-example`. The code is in `Main.java`, the turtle file is `example.ttl`. I used Java 17 for this example.  

I tried using `org.apache.jena:jena-ontapi` in `Newlib.java`, I could not find a way to access this library's Restriction (`org.apache.jena.ontapi.model.OntClass.Restriction`) from the objects I can find.

**Comment by namedgraph** [comment, marked as answer]
@AndreasMeier12 try importing DCTerms into your ontology.

**Reply by sszuev** [reply]
or explicitly declare `dcterms:conformsTo` as a Data/Object Property (new API) or an RDFS Property (legacy API)

**Comment by AndreasMeier12** [comment]
Thank you for your answers! This solved my issue.
